### PR TITLE
Apply address conversion to remaining tests

### PR DESCRIPTION
**Pull request by domob1812**
This applies the new address conversion functionality (from #156) to all remaining tests.  With this, there are no longer any of the manually converted addresses in `electrum_nmc/electrum/tests`.

Note that most of this PR is straight-forward conversion.  But it also contains some extensions to the supported formats in `frombtc` itself.  The changes to `test_storage_upgrade.py` use a new function that recursively converts addresses in a JSON object (rather than manually apply `frombtc`).  This has the effect that we now convert *all* addresses, while previously only those were converted that were needed to make the test pass.  The new behaviour is probably what we want, though.

### CODE REVIEW

**Comment by domob1812** [comment]
Seems I forgot to update the rebranding check again, fixing it now.

**Comment by JeremyRand** [comment]
Thanks, I'll review this ASAP.

**Comment by JeremyRand** [comment]
ACK d45b613ab47c505f4466c228b36031e0a298753b.  I have not tested the code, but I have reviewed it and it looks OK, I agree it can be merged.  I haven't yet looked at the following commits.

File: electrum_nmc/electrum/tests/test_util.py [hunk 1]
```diff
@@ -122,14 +92,10 @@ def test_parse_URI_invalid_address(self):
         self.assertRaises(BaseException, parse_URI, 'namecoin:invalidaddress')
 
     def test_parse_URI_invalid(self):
-        #self.assertRaises(BaseException, parse_URI, 'notbitcoin:15mKKb2eos1hWa6tisdPwwDC1a5J1y9nma')
-        # Converted to Namecoin using `contrib/convertAddress.py` from Namecoin Core.
-        self.assertRaises(BaseException, parse_URI, 'notnamecoin:N1LgXEXdjF7G37MPzgwyATN6joULz6LfdQ')
+        self.assertRaises(BaseException, parse_URI, frombtc('invalidprotocol:15mKKb2eos1hWa6tisdPwwDC1a5J1y9nma'))
```
Comment (JeremyRand) [inline electrum_nmc/electrum/tests/test_util.py]: Why `invalidprotocol:`?  Seems to me that we should be making `parse_URI` receive a protocol of `notnamecoin:`, so that we'll be sure to catch bugs where it accepts invalid protocols whose substring is a valid protocol.  (Seems like an edge case that's unlikely to be hit, but doesn't hurt to test for it.)

Comment (JeremyRand) [reply electrum_nmc/electrum/tests/test_util.py]: (Which means that `from_btc` might need tweaks to properly convert substrings of the protocol.)

**Comment by JeremyRand** [comment]
ACK 5aaca6f44ecd5bdca9a05d43b29d7f3ff121da91 subject to the above requested change.   I have not tested the code, but I have reviewed it and it looks OK (subject to the above requested change), I agree it can be merged. I haven't yet looked at the following commits.

File: electrum_nmc/electrum/tests/test_util.py (hunk 1 above)
Comment (domob1812) [reply electrum_nmc/electrum/tests/test_util.py]: I just chose `invalidprotocol` because that is the "obvious" intent of the test (i.e. what happens for a protocol that is not the expected one).  Your point about it containing "namecoin" as a substring is valid, though, so I've just changed it back to `notnamecoin:`.  I think we can just explicitly change this one (and let `frombtc` just convert the address), instead of adding a special case.

**Comment by domob1812** [comment]
@JeremyRand, thanks for the review so far!  I've addressed your comment.

**Comment by JeremyRand** [comment]
ACK ba098892d9011e0ec301fe43666cfe550d655f74.   I have not tested the code, but I have reviewed it and it looks OK, I agree it can be merged. I haven't yet looked at the following commits.

**Comment by JeremyRand** [comment]
ACK cc0bcff56b8365511dc03c78abc6bc71700db80d.  I have not tested the code, but I have reviewed it and it looks OK, I agree it can be merged. I haven't yet looked at the following commits.

File: contrib/check_nmc_rebranding [hunk 2]
```diff
@@ -9,8 +9,10 @@ if [ -d "./electrum" ]; then
 fi
 
 for i in **/*.py; do
-    # This RegEx is designed to exclude "btchip" and "C2YxWaiTpNPuxjSNhUlBtcJmcdkz3"
-    if ./contrib/print_python_strings "$i" "btc$|btc[^hj]|btch[^i]|btchi[^p]|btcj[^m]|btcjm[^c]"; then
+    if ./contrib/print_python_strings "$i" "btc" \
+        | grep -v "btchip" \
+        | grep -v "C2YxWaiTpNPuxjSNhUlBtcJmcdkz3" \
+        | grep -v "3KaBTcviBLEJajTEMstsA2GWjYoPzPK7Y7"; then
```
Comment (JeremyRand) [inline contrib/check_nmc_rebranding]: What's the motivation for excluding `3KaBTcviBLEJajTEMstsA2GWjYoPzPK7Y7`?  If the intention is to make the unit tests pass, I'd prefer that this exclusion only be applied to the tests, e.g. by matching something like `electrum/tests/.*\.py.*3KaBTcviBLEJajTEMstsA2GWjYoPzPK7Y7`.

**Comment by JeremyRand** [comment]
ACK 58730959346de679f92a80be9ac5084ea90d531f, subject to the above requested change. I have not tested the code, but I have reviewed it and it looks OK (subject to the above requested change), I agree it can be merged.

File: contrib/check_nmc_rebranding (hunk 2 above)
Comment (domob1812) [reply contrib/check_nmc_rebranding]: Yes, the main motivation was because it shows up in tests.  And the motivation for changing this part in general was that the strange, complex regexp with a comment "this excludes ..." is way less readable than just straight-forward excluding those things.  Also the C2YxWaiTpNPuxjSNhUlBtcJmcdkz3 string, which was excluded before globally (indirectly through the regexp), occurs only in tests.

I agree that we should ideally be as restrictive as possible with the exclusions, though, so now both strings are excluded only in the tests.

**Comment by domob1812** [comment]
Updated the rebranding-check as suggested above.

**Comment by JeremyRand** [comment]
ACK 3fa205c1c1e4f68bbfaef9805224cdf21126553a.  I have not tested the code, but I have reviewed it and it looks OK, I agree it can be merged.
